### DISCUSSION TITLE
Handling scrollBehavior

### DISCUSSION THREAD

**Discussion by wilfredinni** [hannoeru/vite-plugin-pages#263] [Q&A]
Hi! How would you handle scrollBehavior with vite-plugin-pages. I need to scroll to top if there is no hash, otherwise, to the hash. Thank you!

**Comment by wilfredinni** [comment]
Never mind, I just added this to `App.vue`

```typescript
const router = useRouter()
router.options.scrollBehavior = (to, _from, savedPosition) => {
  if (savedPosition) return savedPosition
  if (to.hash) return { el: to.hash, behavior: 'smooth' }
  else return { top: 0 }
}
```

**Comment by hannoeru** [comment]
You can setup this when creating vue router instance.

https://router.vuejs.org/guide/advanced/scroll-behavior.html
